roachtest: sqlsmith: to_json(): unexpected type *tree.DEnum for AsJSON

[(roachtest).sqlsmith/setup=seed/setting=no-mutations failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2030392&tab=buildLog) on [master@6b9c32bf8d8553be6b0540f82ce18327b88262b5](https://github.com/cockroachdb/cockroach/commits/6b9c32bf8d8553be6b0540f82ce18327b88262b5):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed/setting=no-mutations/run_1
	sqlsmith.go:169,sqlsmith.go:199,test_runner.go:753: error: pq: internal error: to_json(): unexpected type *tree.DEnum for AsJSON
		stmt:
		SELECT
			to_json(tab_5518._enum)::JSONB AS col_13872,
			tab_5516._int8 AS col_13873,
			tab_5516._date AS col_13874,
			tab_5517._uuid AS col_13875,
			tab_5517._interval AS col_13876,
			NULL AS col_13877,
			5782046592603117708:::INT8 AS col_13878,
			NULL AS col_13879,
			NULL AS col_13880
		FROM
			defaultdb.public.seed@seed__int8__float8__date_idx AS tab_5516,
			defaultdb.public.seed@seed__int8__float8__date_idx AS tab_5517
			JOIN defaultdb.public.seed@seed__int8__float8__date_idx AS tab_5518 ON
					(tab_5517._bool) = (tab_5518._bool) AND (tab_5517._decimal) = (tab_5518._decimal)
			JOIN defaultdb.public.seed AS tab_5519
				JOIN defaultdb.public.seed AS tab_5520 ON
						(tab_5519._int8) = (tab_5520._int8)
						AND (tab_5519._float8) = (tab_5520._float8)
						AND (tab_5519._date) = (tab_5520._date) ON (tab_5517._date) = (tab_5520._date)
		LIMIT
			20:::INT8;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=seed/setting=no-mutations](https://teamcity.cockroachdb.com/viewLog.html?buildId=2030392&tab=artifacts#/sqlsmith/setup=seed/setting=no-mutations)
Related:
- #49251 roachtest: sqlsmith/setup=seed/setting=no-mutations failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dseed%2Fsetting%3Dno-mutations.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/testdata/logic_test/enums
+++ b/pkg/sql/logictest/testdata/logic_test/enums
@@ -1059,3 +1059,8 @@ query T
 SELECT * FROM enum_data_type
 ----
 hello
+
+query T
+SELECT to_json('hello'::greeting)
+----
+"hello"

--- a/pkg/sql/sem/tree/datum.go
+++ b/pkg/sql/sem/tree/datum.go
@@ -2980,6 +2980,8 @@ func AsJSON(d Datum, loc *time.Location) (json.JSON, error) {
 		return json.FromString(string(*t)), nil
 	case *DCollatedString:
 		return json.FromString(t.Contents), nil
+	case *DEnum:
+		return json.FromString(t.LogicalRep), nil
 	case *DJSON:
 		return t.JSON, nil
 	case *DArray:

